packages/expo-modules-core/ios/Swift/Functions/ConcreteFunction.swift

```diff
@@ -90,21 +90,35 @@ public final class ConcreteFunction<Args, ReturnType>: AnyFunction {
 
   private func castArguments(_ args: [Any]) throws -> [Any] {
     if args.count != argumentsCount {
-      throw InvalidArgsNumberError(received: args.count, expected: argumentsCount)
+      throw InvalidArgsNumberException((received: args.count, expected: argumentsCount))
     }
     return try args.enumerated().map { index, arg in
       let expectedType = argumentType(atIndex: index)
 
-      // It's safe to unwrap since the arguments count matches.
-      return try expectedType!.cast(arg)
+      do {
+        // It's safe to unwrap since the arguments count matches.
+        return try expectedType!.cast(arg)
+      } catch {
+        throw ArgumentCastException((index: index, type: expectedType!)).causedBy(error)
+      }
     }
   }
 }
 
-internal struct InvalidArgsNumberError: CodedError {
-  let received: Int
-  let expected: Int
-  var description: String {
-    "Received \(received) arguments, but \(expected) was expected."
+internal class InvalidArgsNumberException: GenericException<(received: Int, expected: Int)> {
+  override var reason: String {
+    "Received \(params.received) arguments, but \(params.expected) was expected."
+  }
+}
+
+internal class ArgumentCastException: GenericException<(index: Int, type: AnyArgumentType)> {
+  override var reason: String {
+    "Argument at index \"\(params.index)\" couldn't be casted to type \"\(params.type.description)\"."
```

Comment:
I don't want to change all the quotes in this PR. I planned to do this once and separately, because there are also some errors/exceptions in the other places that I didn't touch in this PR.

Reply:
casted -> cast